### PR TITLE
fail safely on `teletype add` if the current directory is not a teletype app

### LINKED ISSUE #62: Error with add config
```
C:/Ruby26-x64/lib/ruby/gems/2.6.0/gems/tty-0.10.0/lib/tty/commands/add.rb:69:in `read': No such file or directory @ rb_sysopen - li
b/rev2test/cli.rb (Errno::ENOENT)
```

### PR DESCRIPTION
**Pull request by slowbro**
### Describe the change
Adds a `validate_pwd` method to the "add" command that makes sure that a gemspec & directory exists before trying to create the command.

### Why are we doing this?
A friendly message is better than a stack trace.

### Benefits
This will improve user understanding when accidentally trying to run `teletype add` outside of the project directory.

Should close #62 

### Drawbacks
There's no way to directly check that the pwd is actually a teletype app, vs another app that happens to have the same structure. However, that was the case before too - this just covers one edge case.

### Requirements
Put an X between brackets on each line if you have done the item:
[...] Tests written & passing locally?
[x] Code style checked?
[x] Rebased with `master` branch?
[x] Documentaion updated?

### CODE REVIEW

File: lib/tty/commands/add.rb [hunk 1]
```diff
@@ -73,7 +75,7 @@ def execute(input: $stdin, output: $stdout)
         cmd_integ_test_file = "#{test_dir}/integration/#{cmd_name_path}_#{test_dir}.rb"
         cmd_unit_test_file = "#{test_dir}/unit/#{cmd_name_path}_#{test_dir}.rb"
 
-        if !subcmd_present?
+        unless subcmd_present?
```
[the remark this answers is not in the document]
Comment (slowbro) [reply lib/tty/commands/add.rb:78]: I dunno, I wholeheartedly disagree with this. I feel like the reason `unless` exists is to *not* have `if !...` ... @piotrmurach, what's your take?

This isn't directly related to this change, it was just a thing I modified while running though the file - I'll revert it if needbe.

[the remark this answers is not in the document]
Comment (piotrmurach) [reply lib/tty/commands/add.rb:78]: I don't have a strong preference either way. This change is fine by me.

File: lib/tty/commands/add.rb [hunk 2]
```diff
@@ -272,6 +274,14 @@ def cmd_constantinized_parts
         ].compact
       end
 
+      # Make sure the current directory is a teletype project
+      # (or, at the very least, contains the correct directory structure)
+      #
+      def validate_pwd
+        fail ::TTY::CLI::Error, @pastel.red("This doesn't look like a teletype app directory - are you in the right place?") unless
+          options[:force] || (@app_path + "lib/#{namespaced_path}").exist?
+      end
```
Comment (piotrmurach) [inline lib/tty/commands/add.rb:283]: Since we add commands to `lib/app/commands` folder, maybe we should check that this location is present? After displaying the error message, we could potentially advise on how to run the command with an example or similar to Rails automatically run `teletype add --help` ?

**Review by piotrmurach — COMMENTED** [review]
This will definitely help with user experience. I made a couple of suggestions that may further guide the user, see what you think?

File: lib/tty/commands/add.rb (hunk 2 above)
Comment (slowbro) [reply lib/tty/commands/add.rb:283]: I originally was checking that, yes - but it was causing other tests to fail (they set up a fixture that did not make a 'commands' dir). The add command creates the 'commands' directory if it doesn't exist (seemingly):

```
[katelyn@h test]$ rm -r lib/test/commands
[katelyn@h test]$ teletype add haha
      create  spec/integration/haha_spec.rb
      create  spec/unit/haha_spec.rb
      create  lib/test/commands/haha.rb
      create  lib/test/templates/haha/.gitkeep
      inject  lib/test/cli.rb
[katelyn@h test]$ ls -l lib/test/commands
total 4
-rw-r--r-- 1 katelyn katelyn 332 Sep 28 15:07 haha.rb
```
..so I don't think the check is necessary.

I like the idea of outputting the 'help' - yeah. I'll add that!

**Comment by piotrmurach** [comment]
Thank you! ❤️  My plan is to cut a new release this month.
